src/contracts/atlas/AtlasVerification.sol

```diff
@@ -537,22 +480,23 @@ contract AtlasVerification is EIP712, DAppIntegration {
         payload = _hashTypedDataV4(_getProofHash(userOp));
     }
 
-    function getNextNonce(address account) external view returns (uint256 nextNonce) {
-        NonceTracker memory nonceTracker = asyncNonceBitIndex[account];
+    // Returns the next nonce for the given account, in sequential or async mode
+    function getNextNonce(address account, bool sequenced) external view returns (uint256) {
+        NonceTracker memory nonceTracker = nonceTrackers[account];
 
-        uint256 nextBitmapIndex = uint256(nonceTracker.HighestFullBitmap) + 1;
-        uint256 lowestEmptyBitmap = uint256(nonceTracker.LowestEmptyBitmap);
+        if (sequenced) {
+            return nonceTracker.lastUsedSeqNonce + 1;
+        } else {
+            // Async bitmaps start at index 1. I.e. accounts start with bitmap 0 = HighestFullAsyncBitmap
+            bytes32 bitmapKey = keccak256(abi.encode(account, nonceTracker.highestFullAsyncBitmap + 1));
```

Comment:
As mentioned above, I'm concerned about the FULL_BITMAP check being the only way to increase the highestFullAsyncBitmap

If a nonce is 'missed' in an otherwise full bitmap then the FULL_BITMAP check would fail.. but the getNextNonce func would still try to return a nonce equal to highest nonce + 1, which could exceed 240.

Perhaps we should rename the ``highestFullAsyncBitmap`` to something else.?